auth/permission_details.cc

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#include "auth/permission_details.hh"
+
+// FMT.
+#include <fmt/format.h>
+
+// Scylla includes.
+#include "auth/permission.hh"
+#include "auth/resource.hh"
+#include "cql3/util.hh"
+#include "log.hh"
+
+// STD.
+#include <ranges>
+#include <string_view>
+
+// Note: The name of a keyspace or a table can only consist of alphanumeric types.
+
+namespace auth {
+
+namespace {
+
+logging::logger pdlogger{"permission_details"};
+
+sstring describe_data(std::string_view permission, const resource& r, std::string_view role) {
```

Comment:
I used it to not repeat the same line of code in every function. I'll replace `permissions::to_strings(const permission_set&)` with calls to `permissions::to_string(permission)` to avoid unnecessary allocations

Reply:
> I used it to not repeat the same line of code in every function.

Which line of code? `permissions::to_string`?

> I'll replace `permissions::to_strings(const permission_set&)` with calls to `permissions::to_string(permission)` to avoid unnecessary allocations

My concern is not about performance but rather about type safety. This is not a performance sensitive code path.

As a consolation I'd like to point your attention to the fact that permission names are short strings, so they are eligible for the short string optimization and there will be no allocations ;)